rotkehlchen/data_migrations/migrations/migrations_12.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from typing import TYPE_CHECKING
+
+from rotkehlchen.data_migrations.utils import detect_accounts
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ExternalService, SupportedBlockchain
+
+if TYPE_CHECKING:
+    from rotkehlchen.data_migrations.progress import MigrationProgressHandler
+    from rotkehlchen.rotkehlchen import Rotkehlchen
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def data_migration_12(rotki: 'Rotkehlchen', progress_handler: 'MigrationProgressHandler') -> None:
+    """
+    Introduced at v1.31.0
+    """
+    log.debug('Enter data_migration_12')
+
+    # steps are: ethereum accounts + 3 (potentially write to db + updating spam assets and rpc nodes + new round msg)  # noqa: E501
+    progress_handler.set_total_steps(len(rotki.chains_aggregator.accounts.eth) + 3)
+    detect_accounts(
+        chain=SupportedBlockchain.GNOSIS,
+        external_service=ExternalService.GNOSIS_ETHERSCAN,
+        service_api_key='J3XEY27VIT7377G34PVPHKWG74NG9PXNSM',  # also used in tests
+        rotki=rotki,
+        progress_handler=progress_handler,
```

Comment:
this is missing also the base chain detection but we can add in other PR if this one is fine already

Reply:
Ok, I added base accounts detection too